### PR TITLE
Add move object functions

### PR DESCRIPTION
**Pull request by FerrariDG**
This PR adds the move module that allows moving S3 objects.

### CODE REVIEW

File: s3_tools/read.py [hunk 1]
```diff
@@ -36,12 +33,14 @@ def read_object_to_bytes(bucket: str, key: str) -> Optional[bytes]:
     session = boto3.session.Session()
     s3 = session.client("s3")
     obj = s3.get_object(Bucket=bucket, Key=key)
-    data = obj["Body"].read() if "Body" in obj else None
 
-    return data
+    if "Body" not in obj:
+        raise KeyError("Missing key 'Body' on response")
```
Comment (cmalves) [inline s3_tools/read.py:38]: We can enrich our docstring with a new `Raises` section. [Numpy Style Guide](https://numpydoc.readthedocs.io/en/latest/format.html) 

The next step is to create our custom `Exceptions` right? :slightly_smiling_face:

Comment (FerrariDG) [reply s3_tools/read.py:43]: Added the exceptions to docstring. We can discuss the exceptions and how to deal with them. I have no ideas for now.

File: s3_tools/move.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    threads: int = 5
+) -> None:
+    """Move a list of S3 objects from source bucket to destination.
+
+    Parameters
+    ----------
+    source_bucket : str
+        S3 bucket where the objects are stored.
+
+    source_key : List[str]
```
Comment (cmalves) [inline s3_tools/move.py:63]:
```suggestion
    source_keys : List[str]
```

File: s3_tools/move.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        S3 keys where the objects are referenced.
+
+    destination_bucket : str
+        S3 destination bucket.
+
+    destination_key : List[str]
```
Comment (cmalves) [inline s3_tools/move.py:69]:
```suggestion
    destination_keys : List[str]
```

File: s3_tools/move.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        raise ValueError("Key list length must be greater than zero")
+
+    with futures.ThreadPoolExecutor(max_workers=threads) as executor:
+        executors = (
+            executor.submit(move_object, source_bucket, source, destination_bucket, destination)
+            for source, destination in zip(source_keys, destination_keys)
+        )
+
+        for ex in executors:
+            ex.result()
```
Comment (cmalves) [inline s3_tools/move.py:111]: Why do we need this?

**Review by cmalves — APPROVED** [review]
:white_check_mark:

File: s3_tools/move.py (hunk 4 above)
Comment (FerrariDG) [reply s3_tools/move.py:111]: I'm not sure why but without this the unit test for this method fails. My assumption is that the `submit` is lazy and the threads are really called with the `results` (but it's a wild guess didn't look into it).

Comment (cmalves) [reply s3_tools/move.py:111]: Okok thanks :slightly_smiling_face:

**Pull request merged**